include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -5889,6 +5898,27 @@ struct OrtCompileApi {
 ORT_RUNTIME_CLASS(Ep);
 ORT_RUNTIME_CLASS(EpFactory);
 
+struct OrtEpApi {
+  /** \brief Create an OrtEpDevice for the EP and an OrtHardwareDevice.
+   * \param[in] ep_factory Execution provider factory that is creating the instance.
+   * \param[in] hardware_device Hardware device that the EP can utilize.
+   * \param[in] ep_metadata Optional OrtKeyValuePairs instance for execution provider metadata that may be used
+   *                        during execution provider selection and passed to CreateEp.
+   *                        ep_device will take ownership of this instance.
+   * \param[in] ep_options  Optional OrtKeyValuePairs instance for execution provider options that will be added
+   *                        to the Session configuration options if the execution provider is selected.
+   *                        ep_device will take ownership of this instance.
+   * \param ep_device OrtExecutionDevice that is created.
+   */
+  ORT_API2_STATUS(CreateEpDevice, _In_ OrtEpFactory* ep_factory,
+                  _In_ const OrtHardwareDevice* hardware_device,
+                  _In_ OrtKeyValuePairs* ep_metadata,
+                  _In_ OrtKeyValuePairs* ep_options,
```

Comment:
I see that these are moved (via std::move) in the implementation. AFAIK, this is not common practice in a C API (as std::move is a C++ construct) and can lead to difficult to debug errors because the object provided by the user has been gutted by `std::move`. I would lean towards passing in const pointers and allowing the implementation to make a copy.

But, if we still want to use std::move, then I think we need to do the following:
- Make these `_InOut_` instead of `_In_`.
- Expand documenetation commet to state that the OrtKeyValuePairs they passed in are now empty (due to the move) and can no longer be used by the caller.